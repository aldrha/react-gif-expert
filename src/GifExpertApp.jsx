import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Power Rangers']);

    const onAddCategory = (onNewCategory) => {
        console.log(onNewCategory);
        if (categories.includes(onNewCategory)) return;

        setCategories([onNewCategory, ...categories]);
    };

    return (
        <>
            <h1 aria-label="title">GiftExpertApp</h1>

            <AddCategory onNewCategory={onAddCategory} />

            <ol>
                {categories.map((category) => (
                    <GifGrid key={category} category={category} />
                ))}
            </ol>
        </>
    );
};
