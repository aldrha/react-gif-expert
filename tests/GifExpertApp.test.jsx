import { fireEvent, screen, render } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
    test('debe de hacer match con el snapshop', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar el titulo', () => {
        render(<GifExpertApp />);

        expect(
            screen.getByRole('heading', { level: 1, name: 'title' })
        ).toBeTruthy();

        expect(screen.getByText('GiftExpertApp')).toBeTruthy();
    });

    test('debe agregar una nueva categoria', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: 'Avatar' } });
        fireEvent.submit(form);

        const titles = screen.getAllByRole('heading', {
            level: 1,
            name: 'category-title',
        });

        expect(titles.length).toBeGreaterThan(1);
    });

    //solucion de alguien mas en el curso

    test('no debe de agregar una categoria si ya existe', () => {
        const category = 'Avatar';
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: category } });
        fireEvent.submit(form);

        fireEvent.input(input, { target: { value: category } });
        fireEvent.submit(form);

        // console.log(titles.name.length).toB;
        expect(screen.getAllByText(category).length).toBe(1);
    });
});
