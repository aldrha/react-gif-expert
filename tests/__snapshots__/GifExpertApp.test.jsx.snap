// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <GifExpertApp /> debe de hacer match con el snapshop 1`] = `
<div>
  <h1
    aria-label="title"
  >
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <ol>
    <h1
      aria-label="category-title"
    >
      Power Rangers
    </h1>
    <h2>
      Cargando ...
    </h2>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
